### PR TITLE
Add fan out to history client for GetReplicationStatus

### PR DESCRIPTION
**Pull request by yiminc**
**What changed?**
Add fan out to history client for GetReplicationStatus

### CODE REVIEW

File: client/history/client.go [hunk 1]
```diff
@@ -1053,21 +1054,34 @@ func (c *clientImpl) GetReplicationStatus(
...
+			lock.Lock()
+			defer lock.Unlock()
+			result.Shards = append(result.Shards, resp.Shards...)
+			return nil
+		})
+	}
+	err = eg.Wait()
 	if err != nil {
 		return nil, err
 	}
```
Comment (wxing1292) [inline client/history/client.go:1082]: say there are 10 response / errors returned
above contains one error, will the other 9 success response be thrown away?

File: client/history/client.go [hunk 2]
```diff
@@ -1053,21 +1054,34 @@ func (c *clientImpl) GetReplicationStatus(
...
+			if err != nil {
+				return err
+			}
+			lock.Lock()
+			defer lock.Unlock()
+			result.Shards = append(result.Shards, resp.Shards...)
+			return nil
+		})
+	}
+	err = eg.Wait()
```
Comment (yux0) [inline client/history/client.go:1079]: no experience with this errgroup. Not sure how it handles errors. Does it fail fast when receiving an error?

File: client/history/client.go [hunk 3]
```diff
@@ -1053,21 +1054,34 @@ func (c *clientImpl) GetReplicationStatus(
...
 	if err != nil {
 		return nil, err
```
Comment (yux0) [inline client/history/client.go:1081]: Should it return what it has in the response?

**Review by yux0 — COMMENTED** [review]
Will the GetReplicationStatus API expose via admin client?

File: client/history/client.go (hunk 2 above)
Comment (yiminc) [reply client/history/client.go:1079]: Yes, it cancels context when encounter the first error.

File: client/history/client.go (hunk 3 above)
Comment (yiminc) [reply client/history/client.go:1081]: I think it does not matter.

File: client/history/client.go (hunk 1 above)
Comment (yiminc) [reply client/history/client.go:1082]: error group cancels context when it sees first error.

Comment (wxing1292) [reply client/history/client.go:1082]: >error group cancels context when it sees first error.

exactly, as long as one error encountered, all other results are affected.
we should not use error group for this use case then

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**